programs/ziti-edge-tunnel/netif_driver/windows/tun.c

```diff
@@ -0,0 +1,262 @@
+static WINTUN_GET_RUNNING_DRIVER_VERSION_FUNC WintunGetRunningDriverVersion;
+static WINTUN_SET_LOGGER_FUNC WintunSetLogger;
+static WINTUN_START_SESSION_FUNC WintunStartSession;
+static WINTUN_END_SESSION_FUNC WintunEndSession;
+static WINTUN_GET_READ_WAIT_EVENT_FUNC WintunGetReadWaitEvent;
+static WINTUN_RECEIVE_PACKET_FUNC WintunReceivePacket;
+static WINTUN_RELEASE_RECEIVE_PACKET_FUNC WintunReleaseReceivePacket;
+static WINTUN_ALLOCATE_SEND_PACKET_FUNC WintunAllocateSendPacket;
+static WINTUN_SEND_PACKET_FUNC WintunSendPacket;
+
+static uv_once_t wintunInit;
+static HMODULE WINTUN;
+
+static void InitializeWintun(void)
+{
+    HMODULE Wintun =
+            LoadLibraryExW(L"wintun.dll", NULL, LOAD_LIBRARY_SEARCH_APPLICATION_DIR | LOAD_LIBRARY_SEARCH_SYSTEM32);
+    if (!Wintun)
+        return;
+#define X(Name, Type) ((Name = (Type)GetProcAddress(Wintun, #Name)) == NULL)
+    if (X(WintunCreateAdapter, WINTUN_CREATE_ADAPTER_FUNC) ||
+        X(WintunDeleteAdapter, WINTUN_DELETE_ADAPTER_FUNC) ||
+        X(WintunDeletePoolDriver, WINTUN_DELETE_POOL_DRIVER_FUNC) ||
+        X(WintunEnumAdapters, WINTUN_ENUM_ADAPTERS_FUNC) ||
+        X(WintunFreeAdapter, WINTUN_FREE_ADAPTER_FUNC) ||
+        X(WintunOpenAdapter, WINTUN_OPEN_ADAPTER_FUNC) ||
+        X(WintunGetAdapterLUID, WINTUN_GET_ADAPTER_LUID_FUNC) ||
+        X(WintunGetAdapterName, WINTUN_GET_ADAPTER_NAME_FUNC) ||
+        X(WintunSetAdapterName, WINTUN_SET_ADAPTER_NAME_FUNC) ||
+        X(WintunGetRunningDriverVersion, WINTUN_GET_RUNNING_DRIVER_VERSION_FUNC) ||
+        X(WintunSetLogger, WINTUN_SET_LOGGER_FUNC) ||
+        X(WintunStartSession, WINTUN_START_SESSION_FUNC) ||
+        X(WintunEndSession, WINTUN_END_SESSION_FUNC) ||
+        X(WintunGetReadWaitEvent, WINTUN_GET_READ_WAIT_EVENT_FUNC) ||
+    X(WintunReceivePacket, WINTUN_RECEIVE_PACKET_FUNC) ||
+    X(WintunReleaseReceivePacket, WINTUN_RELEASE_RECEIVE_PACKET_FUNC) ||
+    X(WintunAllocateSendPacket, WINTUN_ALLOCATE_SEND_PACKET_FUNC) ||
+    X(WintunSendPacket, WINTUN_SEND_PACKET_FUNC))
+#undef X
+    {
+        DWORD LastError = GetLastError();
+        FreeLibrary(Wintun);
+        SetLastError(LastError);
+        Wintun = NULL;
+    }
+
+    WINTUN = Wintun;
+}
+
+netif_driver tun_open(struct uv_loop_s *loop, uint32_t tun_ip, uint32_t dns_ip, const char *cidr, char *error, size_t error_len) {
+    if (error != NULL) {
+        memset(error, 0, error_len * sizeof(char));
+    }
+
+    uv_once(&wintunInit, InitializeWintun);
+    if (WINTUN == NULL) {
+        strcpy_s(error, error_len, "Failed to load wintun.dll");
+        return NULL;
+    }
+
+    struct netif_handle_s *tun = calloc(1, sizeof(struct netif_handle_s));
+    if (tun == NULL) {
+        if (error != NULL) {
+            snprintf(error, error_len, "failed to allocate tun");
+        }
+        return NULL;
+    }
+
+    GUID adapterGuid;
+    IIDFromString(L"2cbfd72d-370c-43b0-b0cd-c8f092a7e134", &adapterGuid);
```

Comment:
I assume you just generated a uuid? I feel like this is better expressed as a constant somewhere with a comment about what it is and why, i expect it's "important"?